lebedigital/openbis/README.md

```diff
@@ -162,4 +162,22 @@ samples = o.get_samples(
 ).df
 ```
 > For detailed information see [here](https://pypi.org/project/PyBIS/) under 'search for samples / objects'.
-> Searching is also possible via Web-View.
\ No newline at end of file
+> Searching is also possible via Web-View.
+
+## Generating and working with type checkers.
+
+> You can generate type-checkers based on pydantic models to check your samples for formatting errors before uploading them to openbis
+
+```python
+sample_properties = {"$name": "sample_name"}
+
+SampleModel = o.generate_typechecker("SAMPLE_TYPE")
+sample_model_return = SampleModel(**sample_properties)
+sample_properties = sample_model_return.dict(exclude_unset=True) # exclude_unset is for not including other property types that were not set in sample_properties
+```
+
+> The sample_properties have been checked for their types and casted/formatted if possible. The functionality includes:
+
+- Casting strings like "2.13" or "7" to float or integer respectively
+- Checking if CONTROLLED_VOCABULARY Properties are withing their defined vocabularies
+- Casting date-strings into the required format. Ex "17.10.2023 10:45" will be formatted to "2023-10-17"
```

Comment:
Thanks for adding that.
What's with the time? Shouldn't it be 2023-10-17 10:45 at the end?

Reply:
Not if the property type is a date, if it was a timestamp then yes it would include the date. The formatting itself is in my opinion not that important though as the main point is that the formatting will be done automatically.